packages/syft/src/syft/service/code_history/code_history_service.py

```diff
@@ -0,0 +1,261 @@
+# stdlib
+from typing import List
+from typing import Optional
+from typing import Union
+
+# relative
+from ...serde.serializable import serializable
+from ...store.document_store import DocumentStore
+from ...types.uid import UID
+from ...util.telemetry import instrument
+from ..code.user_code import SubmitUserCode
+from ..code.user_code import UserCode
+from ..context import AuthedServiceContext
+from ..response import SyftError
+from ..response import SyftSuccess
+from ..service import AbstractService
+from ..service import service_method
+from ..user.user_roles import DATA_OWNER_ROLE_LEVEL
+from ..user.user_roles import DATA_SCIENTIST_ROLE_LEVEL
+from .code_history import CodeHistory
+from .code_history import CodeHistoryDict
+from .code_history import CodeVersions
+from .code_history import UserHistoryDict
+from .code_history_stash import CodeHistoryStash
+
+
+@instrument
+@serializable()
+class CodeHistoryService(AbstractService):
+    store: DocumentStore
+    stash: CodeHistoryStash
+
+    def __init__(self, store: DocumentStore) -> None:
+        self.store = store
+        self.stash = CodeHistoryStash(store=store)
+
+    @service_method(
+        path="code_history.submit_version",
+        name="submit_version",
+        roles=DATA_SCIENTIST_ROLE_LEVEL,
+    )
+    def submit_version(
+        self,
+        context: AuthedServiceContext,
+        code: Union[SubmitUserCode, UserCode],
+        comment: Optional[str] = None,
+        add_request: Optional[bool] = True,
+        reason: Optional[str] = None,
+    ) -> Union[SyftSuccess, SyftError]:
+        user_code_service = context.node.get_service("usercodeservice")
+
+        final_result = None
+        if isinstance(code, SubmitUserCode):
+            if add_request:
+                result = user_code_service.request_code_execution(
+                    context=context, code=code, reason=reason
+                )
+                if isinstance(result, SyftError):
+                    return result
+                code = (
+                    result.ok()
+                    .changes[0]
+                    .linked_obj.resolve_with_context(context=context)
+                    .ok()
+                )
+                final_result = result
+            else:
+                result = user_code_service.submit(context=context, code=code)
+                if isinstance(result, SyftError):
+                    return result
+                uid = UID.from_string(result.message.split(" ")[-1])
```

Comment:
maybe we can access the stash directly or create a intermediate method that does this and returns the object so we dont have to parse the id from the message

Reply:
not sure if I understand this one. What we need here is the code object to validate de UID and the service_func_name, and we try to get that from the server to avoid any attacks. The result of user_code.submit was not returning anything useful which is way we added the id and parsed it in the function. Not sure how to do that without bloating the message result